lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -208,5 +206,151 @@ def test_clear_tools_test70(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    assert another_group.exists() is False
+    assert fubar_another_host.exists() is False
+
+
+def test_clear_tools_test73(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear the tools, by name, separated by comma"""
+    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
+    default_group = pbench_run / "tools-v1-default"
+    default_group.mkdir(parents=True)
+    fubar6_host = default_group / "fubar6.example.com"
+    fubar6_host.mkdir(parents=True, exist_ok=True)
+    pidstat6_tool = fubar6_host / "pidstat"
+    pidstat6_tool.touch()
+    mpstat6_tool = fubar6_host / "mpstat"
+    mpstat6_tool.touch()
+    iostat6_tool = fubar6_host / "iostat"
+    iostat6_tool.touch()
+
+    command = ["pbench-clear-tools", "--name=pidstat,mpstat"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert b"" in err
+    assert pidstat6_tool.exists() is False
+    assert mpstat6_tool.exists() is False
+    assert iostat6_tool.exists() is True
+    assert fubar6_host.exists() is True
+
+
+def test_clear_tools_test74(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear all tools in default group, by name, separated by
+    comma"""
+    monkeypatch.setenv("_PBENCH_AGENT_CONFIG", str(pbench_cfg))
+    default_group = pbench_run / "tools-v1-default"
+    default_group.mkdir(parents=True)
+    fubar6_host = default_group / "fubar6.example.com"
+    fubar6_host.mkdir(parents=True, exist_ok=True)
+    pidstat6_tool = fubar6_host / "pidstat"
+    pidstat6_tool.touch()
+    mpstat6_tool = fubar6_host / "mpstat"
+    mpstat6_tool.touch()
+    iostat6_tool = fubar6_host / "iostat"
+    iostat6_tool.touch()
+
+    command = ["pbench-clear-tools", "--name=pidstat,mpstat,iostat"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert pidstat6_tool.exists() is False
+    assert mpstat6_tool.exists() is False
+    assert iostat6_tool.exists() is False
+    assert fubar6_host.exists() is False
+    assert default_group.exists() is True
+
+
+def test_clear_tools_test75(monkeypatch, agent_config, pbench_run, pbench_cfg):
+    """ Attempt to clear all tools in non-default group, by name, separated by
+        comma"""
+    another_group = pbench_run / "tools-v1-another"
+    another_group.mkdir(parents=True)
+    fubar7_host = another_group / "fubar7.example.com"
+    fubar7_host.mkdir(parents=True, exist_ok=True)
+    pidstat7_tool = fubar7_host / "pidstat"
+    pidstat7_tool.touch()
+    mpstat7_tool = fubar7_host / "mpstat"
+    mpstat7_tool.touch()
+
+    command = ["pbench-clear-tools", "--group=another", "--name=pidstat,mpstat"]
+    out, err, exitcode = pytest.helpers.capture(command)
+    assert b"" == out
+    assert pidstat7_tool.exists() is False
+    assert mpstat7_tool.exists() is False
+    assert fubar7_host.exists() is False
+    assert another_group.exists() is False
```

Comment:
If the group is gone then the host and tools are gone, right?  So, I think we need only check for the absence of the group.